cmake/aws-c-event-stream-config.cmake

```diff
@@ -2,5 +2,9 @@ include(CMakeFindDependencyMacro)
 find_dependency(aws-c-common)
 find_dependency(aws-checksums)
 
-include(${CMAKE_CURRENT_LIST_DIR}/@CMAKE_PROJECT_NAME@-targets.cmake)
+if (BUILD_SHARED_LIBS)
```

Comment:
Here was what I mentioned in the meeting. When compiling customer application, they probably won't specify BUILD_SHARED_LIBS anywhere, because they are likely building executables, so if c libs are installed as shared, they won't be able to find the default static one. Or if they are building shared libs of their own, BUILD_SHARED_LIBS is probably set to on, but if c libs are installed as static, they won't be able to find one neither (This is a more valid issue, since BUILD_SHARED_LIBS is a CMake variable).

I'd use a different variable to differentiate that. like `-DLINKING_SHARED_AWS_LIBS`
```
if (LINKING_SHARED_AWS_LIBS)
    include(shared)
else()
    include(static)
endif()
```